core/state_processor.go

```diff
@@ -344,37 +344,79 @@ func (p *StateProcessor) Process(block *types.Block, etxSet *types.EtxSet) (type
 		i++
 	}
 
-	qiTransactions := block.QiTransactions()
+	coinbaseTx := block.Transactions()[0]
 	// Coinbase check
-	if len(qiTransactions) > 0 && types.IsCoinBaseTx(qiTransactions[0], header.ParentHash(nodeCtx), nodeLocation) {
-		totalCoinbaseOut := big.NewInt(0)
-		for _, txOut := range qiTransactions[0].TxOut() {
-			totalCoinbaseOut.Add(totalCoinbaseOut, types.Denominations[txOut.Denomination])
-		}
-		reward := misc.CalculateReward(header)
-		maxCoinbaseOut := new(big.Int).Add(reward, totalFees) // TODO: Miner tip will soon no longer exist
+	if types.IsCoinBaseTx(coinbaseTx, header.ParentHash(nodeCtx), nodeLocation) {
```

Comment:
I'm not seeing where we enforce that the coinbase TX may only exist as the first TX. e.g. what if a second coinbase TX shows up later in the list?